src/fontra_glyphs/backend.py

```
@@ -640,3 +800,205 @@ def gsVerticalMetricsToFontraLineMetricsHorizontal(gsFont, gsMaster):
+
+                if not masterId:
+                    # If it has glyph axes and is not on any master location,
+                    # it must be an intermediate master.
+                    isIntermediateLayer = True
+                else:
+                    # If it has glyph axes and is on a master location,
+                    # but any of the glyph axes are not at min or max position,
+                    # it must be an intermediate layer.
+                    if any(
+                        [
+                            True
+                            for axis in variableGlyph.axes
+                            if glyphLocation[axis.name]
+                            not in [axis.minValue, axis.maxValue]
+                        ]
+                    ):
+                        isIntermediateLayer = True
+
+                if isIntermediateLayer:
+                    raise NotImplementedError(
+                        "GlyphsApp Backend: Intermediate layers "
+                        "within smart glyphs are not yet implemented."
+                    )
+
+            fontraLayerToGSLayer(layer, gsLayer)
+            gsGlyph.layers.append(gsLayer)
+
+    return gsGlyph
+
+
+def fontraLayerToGSLayer(layer, gsLayer):
+    gsLayer.paths = []
+
+    # Draw new paths with pen
+    pen = gsLayer.getPointPen()
+    layer.glyph.path.drawPoints(pen)
+
+    gsLayer.width = layer.glyph.xAdvance
+    gsLayer.components = [
+        fontraComponentToGSComponent(component) for component in layer.glyph.components
+    ]
+    gsLayer.anchors = [fontraAnchorToGSAnchor(anchor) for anchor in layer.glyph.anchors]
+    gsLayer.guides = [
+        fontraGuidelineToGSGuide(guideline) for guideline in layer.glyph.guidelines
+    ]
+
+
+def fontraComponentToGSComponent(component):
+    gsComponent = glyphsLib.classes.GSComponent(component.name)
+    transformation = component.transformation.toTransform()
+    if not isinstance(transformation, Transform):
```

I got confused between `glyphsLib.types.Transform` and `fontTools.misc.transform.Transform`.

`component.transformation.toTransform()` returns a fonttools `Transform`, so this condition is always `True`.

I would suggest to do this:
```python
from glyphsLib.types import Transform as GSTransform`
```
to avoid any confusion, even if we don't need fonttools's `Transform` here (yet).

That said, it seems that glyphsLib's conversion of the transform does not support skewing. If we apply a spew to a component in Fontra, it is not written as a skewed transform. Not sure this is an oversight in glyphsLib or a limitation of the .glyphs format.

If glyphsLib: we should consider contributing a PR
If limitation of the format: we should raise an error if skew is used. (See fontra-rcjk, it also does not support skew for components)

---

OK, I added a unittest for that. It raises a TypeError (similar to fontra-rcjk) + I opened an issue in glyphsLib: https://github.com/googlefonts/glyphsLib/issues/1067

I am closing this as resolved. Once it has been fixed in glyphsLib, we want to remove the TypeError.